### DISCUSSION TITLE
debug client render on prod

### DISCUSSION THREAD

**Discussion by simplecommerce** [vikejs/vike#1124] [Discussion] [answered]
Hi, is there a way to debug or catch client side rendering import issues when running `server:prod`.

My render hook or page seems to be causing an issue on client side in my react test project.
It runs fine in dev mode but not after i build.
Could be an import issue but there are no logs in the console.

**Comment by brillout** [comment]
Any error thrown by your code should be logged. In other words: shall vite-plugin-ssr swallow an error then I consider it a (critical) bug.

Beyond that, I'm not sure what you want nor what could help you find your issue which seems unrelated to vite-plugin-ssr.

**Reply by simplecommerce** [reply]
Yeah I agree it is an issue in my code, what is weird is I have no issues in dev mode, but when I build it and run it, the client side react render doesn't get executed.

If I comment all my imports in the client render and do a console.log, I see them. So I know its related to my imports.
What is weird is I don't see any console.log errors at all bring thrown for some reason.

**Reply by simplecommerce** [reply]
I did some further testing and for some reason some imports that I have are "hanging" and not returning any errors when this is called: 

`__vitePreload` 

I am excluding some from being bundled in the build using the `vite.config.ts` but I assumed that it would crash or log to the console an issue if the loading wasn't successful, not sure why it doesn't

**Reply by simplecommerce** [reply, marked as answer]
Ok I think I figured out the issue or part of the issue.
I think it is a circular dependency and import loop caused by how the chunks are made.
I was sort of able to make it semi-work by following this: https://github.com/vitejs/vite/discussions/5793#discussioncomment-6535621

I noticed that a file that was being imported by `__vitePreload`, would import another chunk for a common js function helper but that function was in the same chunk that it was trying to import causing a loop.

Its def not an issue with this package. 
Could be also related to this: https://github.com/rollup/rollup/issues/4708

**Reply by brillout** [reply]
👍 Thanks for circling back on this.
